monkey/tests/unit_tests/monkey_island/cc/repository/test_mongo_event_repository.py

```diff
@@ -0,0 +1,177 @@
+
+    return mongo_client
+
+
+@pytest.fixture
+def error_raising_mongo_repository(
+    error_raising_mongo_client, event_serializer_registry
+) -> IEventRepository:
+    return MongoEventRepository(error_raising_mongo_client, event_serializer_registry)
+
+
+def assert_same_contents(a, b):
+    assert len(a) == len(b)
+    for item in a:
+        assert item in b
+
+
+def test_mongo_event_repository__save_event(mongo_repository: IEventRepository):
+    event = FakeAgentEvent(source=uuid.uuid4())
+    mongo_repository.save_event(event)
+    events = mongo_repository.get_events()
+
+    assert event in events
+
+
+def test_mongo_event_repository__save_event_raises(
+    error_raising_mongo_repository: IEventRepository,
+):
+    event = FakeAgentEvent(source=uuid.uuid4())
+
+    with pytest.raises(StorageError):
+        error_raising_mongo_repository.save_event(event)
+
+
+def test_mongo_event_repository__get_events(mongo_repository: IEventRepository):
+    events = mongo_repository.get_events()
+
+    assert_same_contents(events, EVENTS)
+
+
+def test_mongo_event_repository__get_events_raises(
+    error_raising_mongo_repository: IEventRepository,
+):
+    with pytest.raises(RetrievalError):
+        error_raising_mongo_repository.get_events()
+
+
+def test_mongo_event_repository__get_events_by_type(mongo_repository: IEventRepository):
+    events = mongo_repository.get_events_by_type(FakeAgentItemEvent)
+
+    expected_events = [EVENTS[3]]
+    assert_same_contents(events, expected_events)
+
+
+def test_mongo_event_repository__get_events_by_type_raises(
+    error_raising_mongo_repository: IEventRepository,
+):
+    with pytest.raises(RetrievalError):
+        error_raising_mongo_repository.get_events_by_type(FakeAgentItemEvent)
+
+
+def test_mongo_event_repository__get_events_by_tag(mongo_repository: IEventRepository):
+    events = mongo_repository.get_events_by_tag("bar")
+
+    expected_events = [EVENTS[1], EVENTS[2]]
+    assert_same_contents(events, expected_events)
+
+
+def test_mongo_event_repository__get_events_by_tag_raises(
+    error_raising_mongo_repository: IEventRepository,
+):
+    with pytest.raises(RetrievalError):
+        error_raising_mongo_repository.get_events_by_tag("bar")
+
+
+def test_mongo_event_repository__get_events_by_source(mongo_repository: IEventRepository):
+    source_event = EVENTS[2]
+    events = mongo_repository.get_events_by_source(source_event.source)
+
+    expected_events = [source_event]
+    assert_same_contents(events, expected_events)
+
+
+def test_mongo_event_repository__get_events_by_source_raises(
+    error_raising_mongo_repository: IEventRepository,
+):
+    with pytest.raises(RetrievalError):
+        source_event = EVENTS[2]
+        error_raising_mongo_repository.get_events_by_source(source_event.source)
+
+
+def test_mongo_event_repository__reset(mongo_repository: IEventRepository):
+    mongo_repository.reset()
+    events = mongo_repository.get_events()
+
+    assert not events
```

Comment:
I like to test that the repository has something, then reset, then test that it's empty. Maybe it's overkill, but it prevents the test from passing erroniously.

Reply:
The fixture already has all of the events from `EVENTS`. Or would you like events to be explicitly added in the test?

Reply:
I prefer to explicitly `get_events()`, make sure it's non-zero, and then call `reset()`.

Reply:
Thanks for the clarification!